### ISSUE TITLE
Indexing terms other than rdfs:label

### ISSUE THREAD

**Issue by trapperkeeper** [cygri/vocidex#8]
Is there a way to configure vocidex to index on other terms (annotation property values) attached to the classes? I've got the basic set-up running, and it works very well. But I can't figure out how to get it to index values other than rdfs:label. Sorry, I know that this isn't an "issue" per se. But I didn't find any other contact method.

**Comment by cygri** [comment]
The property to be used can't be changed from configuration, but it would be a fairly simple change in the source code. I suppose you already have Vocidex compiled from source with Maven?

**Comment by trapperkeeper** [comment]
Yes, I did compile with Maven. Could you give me a hint of what 
class(es) I should look at if I want to grab terms from other properties?
Thanks,
Todd

Landon Todd Detwiler
Structural Informatics Group (SIG)
University of Washington

phone: 206-351-7721

**Comment by cygri** [comment]
Check out `LabelDescriber.java` in the `org.deri.vocidex.describers` package. The way it works is, it runs the SPARQL query in `src/main/resources/queries/term-label.sparql` against the input, and uses the first result for `?label` as the label. You can modify that query to use other or additional properties besides `rdfs:label`. Any post-processing is done again in `LabelDescriber`. Hope that helps!

**Comment by trapperkeeper** [comment]
It sounds like the query could be altered fairly easily to return more 
terms, but I'd also want to change the code to index by more than 1 term 
per class. Doable I'm sure. I'm also looking into SPARQL engines that 
have term indexing built in. Thanks for the follow-up.
Cheers,
Todd

Landon Todd Detwiler
Structural Informatics Group (SIG)
University of Washington

phone: 206-351-7721
